app/presenters/curation_concerns/inspect_work_presenter.rb

```diff
@@ -0,0 +1,64 @@
+module CurationConcerns
+  class InspectWorkPresenter
+    def initialize(solr_document, current_ability)
+      @solr_document = solr_document
+      @current_ability = current_ability
+    end
+    attr_reader :solr_document, :current_ability
+
+    def workflow
+      if sipity_entity
+        {
+          entity_id: sipity_entity.id,
+          proxy_for: sipity_entity.proxy_for_global_id,
+          workflow_id: sipity_entity.workflow_id,
+          workflow_name: sipity_entity.workflow_name,
+          state_id: sipity_entity.workflow_state_id,
+          state_name: sipity_entity.workflow_state_name,
+          comments: workflow_comments,
+          roles: sipity_role # TODO: How to get related roles and actors for the sipity_entity
+        }
+      end
+    end
+
+    def solr
+      work.to_solr if work
+    end
+
+    def storage
+      work.inspect if work
+    end
+
+    private
+
+      def sipity_entity
+        @sipity_entity ||= PowerConverter.convert(solr_document, to: :sipity_entity)
+      rescue PowerConverter::ConversionError
+        nil
+      end
+
+      def sipity_role
+        # TODO: How to use PowerConverter to get info on roles for a sipity_entity
+        'Not impmlemented'
+      end
+
+      def sipity_actor
+        # TODO: How to use PowerConverter to get info on actors for a sipity_role
+      end
+
+      def workflow_comments
+        return [] unless sipity_entity && sipity_entity.comments.count > 0
+        comments = []
+        sipity_entity.comments.each do |comment|
+          comments << { comment: comment.comment, created_at: comment.created_at }
+        end
+        comments
+      end
+
+      def work
+        # This presenter just has a solr_document but we might want to inspect the work object
+        # TODO: Is there a better way at this point to get back to the actual work object
```

Comment:
There is no better way, but you should just be able to inspect the solr_doc. I can't see any reason to load the work from fedora (it will be slower)